### ISSUE TITLE
Hitting return on the plant species grid submits the whole form

### ISSUE THREAD

**Issue by Sam-Amy** [BiologicalRecordsCentre/iRecord#1675]
This is an issue @sacrevert has found with the BSBI vascular plants form. Pressing return (i.e. enter) on the keyboard whilst in the species grid submitted the entire form, which is annoying if only half way through compiling.

I've also tested on the standard ‘enter a list of records’ for, and the same happens there, but only if you have already been to the second tab to enter the location information, otherwise you get an error prompting you to fill in the requred field on that second tab, so perhaps nt an issues here (@kitenetter, do you agree?). 

Is there any way to prevent this on the [vascular plant form](https://irecord.org.uk/enter-vascular-plants)?

**Comment by kitenetter** [comment]
@Sam-Amy I'm not aware of people having problems with hitting Enter when using the standard add a list form, so I agree it's not a major issue for that (although it does seem like undesirable behaviour).

**Comment by johnvanbreda** [comment]
Its standard behaviour on a form on a web page - pressing enter in any control inside the form will submit the form.

**Comment by Sam-Amy** [comment]
OK, thanks @johnvanbreda. Perhaps we could consider adding a warning about this @sacrevert to the page of the plant form?

**Comment by sacrevert** [comment]
I suppose most webforms have a higher proportion of mandatory fields though, and a smaller total number of fields. If subscribe to an email or buy something, it's quick to complete and important detail will be mandatory. Here we may have tens of rows (especially for plants and bryos at 1 or 2km), not mandatory, and pressing enter accidentally is very annoying. Could we at least have a popup box to confirm submission if submission is triggered by pressing return? @johnvanbreda

**Comment by johnvanbreda** [comment]
Although it is standard practice, I agree it's not the expected behaviour when inside the grid. Perhaps it would be better for the return key to move to the next row - i.e. a bit like you are submitting just the row you are on, rather than the whole form.

**Comment by sacrevert** [comment]
Yes, I think that is exactly what happened to me. There is also the potential confusion when in comments boxes or free text fields, as in some cases (don't know about iRecord) some websites allow carriage returns within free text fields.

Who is this best assigned to, and shall i change title to cover all such grids including the standard 'enter list of records'?

**Comment by sacrevert** [comment]
Note that this also applies to https://www.brc.ac.uk/irecord/enter_bryophyte_records
as well as https://irecord.org.uk/enter-vascular-plants

@MarkAlliston Is this something that you would be able to take on?

**Comment by MarkAlliston** [comment]
Yes, I have assigned it to myself so will schedule it in.

**Comment by MarkAlliston** [comment]
I have added some javascript to the https://irecord.org.uk/enter-vascular-plants page thats stops enter submitting the form by default.   Once confirmed I can copy to other forms or add it to main javascriipt library so it applies to all forms.

**Comment by sacrevert** [comment]
Seems to work at this end. Can you add to this form as well please: https://www.brc.ac.uk/irecord/enter_bryophyte_records

I'm not sure about applying it to other forms; this would be something for @kitenetter

**Comment by MarkAlliston** [comment]
I have added it to the Bryophyte page as well.  
If anyone amends the description on the page they will need to check they have have not removed the script.  ( The source tab on the description )
